project/Build.scala

```diff
@@ -190,7 +190,7 @@ object Build extends sbt.Build {
           "org.mockito" % "mockito-core" % mockitoVersion % "test"
         )
       )
-  ) dependsOn(core % "test->test", streaming % "provided", external_kafka  % "provided")
+  ) dependsOn(streaming % "test->test;compile->compile", streaming % "provided", external_kafka  % "provided")
```

Comment:
have you verified the assembly jar should exclude streaming classes and its dependencies ?